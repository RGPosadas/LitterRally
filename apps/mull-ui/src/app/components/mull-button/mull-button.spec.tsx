import { render } from '@testing-library/react';
import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import MullButton from './mull-button';

describe('MullButton', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<MullButton />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<MullButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should do nothing on click by default', () => {
    const dom = render(<MullButton />);

    const button = dom.getByTestId('mull-button');
    ReactTestUtils.Simulate.click(button);

    expect(button.onclick.call.length).toBe(1);
  });

  it('should run the provided function on click', () => {
    const mockCallback = jest.fn(() => {
      /* Do nothing */
    });
    const dom = render(<MullButton onClick={mockCallback} />);

    const button = dom.getByTestId('mull-button');
    ReactTestUtils.Simulate.click(button);

    expect(mockCallback.mock.calls.length).toBe(1);
  });
});
