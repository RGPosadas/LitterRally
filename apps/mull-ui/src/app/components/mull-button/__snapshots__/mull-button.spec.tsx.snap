// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MullButton should match snapshot 1`] = `
<button
  className="mull-button  "
  data-testid="mull-button"
  onClick={[Function]}
  type="submit"
>
  
</button>
`;
